lib/jekyll/polyglot/patches/jekyll/site.rb

```diff
@@ -43,7 +43,9 @@ def process
             while pids.length >= (lang == all_langs[-1] ? 1 : nproc)
               sleep 0.1
               pids.map do |lang, pid|
-                pids.delete lang if waitpid pid, Process::WNOHANG
+                next unless waitpid pid, Process::WNOHANG
+                pids.delete lang
+                raise Errors::FatalException.new "subprocess #{pid} (#{lang}) failed (#{$?.exitstatus})" unless $?.success?
```

Comment:
I'm learning some new ruby, `$?` is crafty

Reply:
can we get a test for this? something that checks a site build will raise `FatalException` if a build fails. let me know if you need further assistance.

Reply:
I'll look at writing a test when I have some time :)